### PR TITLE
Add esp_encrypted_img to component manager

### PR DESCRIPTION
**Pull request by hmalpani**
Add esp_encrypted_img component to component manager

### CODE REVIEW

File: esp_encrypted_img/idf_component.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+dependencies:
+  idf:
+    version: '>=5.0'
```
Comment (mahavirj) [inline esp_encrypted_img/idf_component.yml:3]:
```suggestion
    version: '>=4.3'
```

This is standalone component, should work with previous IDF releases as well?

File: esp_encrypted_img/idf_component.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+dependencies:
```
Comment (mahavirj) [inline esp_encrypted_img/idf_component.yml:1]: Please reorder this file per https://github.com/espressif/idf-extra-components/blob/master/sh2lib/idf_component.yml for consistency

File: esp_encrypted_img/README.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Encrypted Image Decryptor component
```
Comment (mahavirj) [inline esp_encrypted_img/README.md:1]:
```suggestion
# ESP Encrypted Image Abstraction Layer
```

For consistency with component yml file

File: esp_encrypted_img/README.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+
+This component is used for decryption of encrypted images used during OTA updates.
```
Comment (mahavirj) [inline esp_encrypted_img/README.md:3]:
```suggestion
This component provides an API interface to decrypt data defined in "ESP Encrypted Image" format. This format is as specified below:
```

This component can help in integrating pre encrypted firmware in over-the-air updates. Additionally, this component can also be used for other use-cases which requires addition of encryption layer for custom data.

Please add relevant image format diagram here.

File: esp_encrypted_img/README.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+This component also contains tool to generate encrypted image which can be used during OTA updates.
```
Comment (mahavirj) [inline esp_encrypted_img/README.md:5]:
```suggestion
This component also contains tool to generate encrypted images using RSA3072 public key.
```

File: esp_encrypted_img/README.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+
+This component is used as part of following ESP-IDF examples:
```
Comment (mahavirj) [inline esp_encrypted_img/README.md:7]: We can remove this line. We can keep reference to this component from IDF example README as that will be starting point for users.

File: esp_encrypted_img/README.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+- [pre_encrypted_ota](https://github.com/espressif/esp-idf/tree/master/examples/system/ota/pre_encrypted_ota)
+
+To learn more about how to use this component, please check API Documentation from header file.
```
Comment (mahavirj) [inline esp_encrypted_img/README.md:10]: Please add path to header file here.

**Review by mahavirj — CHANGES_REQUESTED** [review]

File: esp_encrypted_img/idf_component.yml [hunk 8]
```diff
@@ -0,0 +1,6 @@
...
+  idf:
+    version: '>=5.0'
+description: ESP Encrypted Image Abstraction Layer
+url: https://github.com/espressif/idf-extra-components/tree/master/esp_encrypted_img
+version: "0.1.0"
```
Comment (mahavirj) [inline esp_encrypted_img/idf_component.yml:6]:
```suggestion
version: "1.0.0"
```

**Review by mahavirj — CHANGES_REQUESTED** [review]

File: esp_encrypted_img/README.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
+# ESP Encrypted Image Abstraction Layer
+
+This component provides an API interface to decrypt data defined in "ESP Encrypted Image" format. This format is as specified at [Image Format](#image-format)
+
+This component can help in integrating pre encrypted firmware in over-the-air updates. Additionally, this component can also be used for other use-cases which requires addition of encryption layer for custom data.
+
+
+## Image Format
+
+![Image Format](img_format.png)
```
Comment (mahavirj) [inline esp_encrypted_img/README.md:10]: File name looks incorrect here, can you please check?

File: test_app/CMakeLists.txt [hunk 10]
```diff
@@ -2,10 +2,10 @@
 # CMakeLists in this exact order for cmake to work correctly
 cmake_minimum_required(VERSION 3.5)
 
-set(EXTRA_COMPONENT_DIRS ../libsodium ../expat)
+set(EXTRA_COMPONENT_DIRS ../libsodium ../expat ../esp_encrypted_img)
```
Comment (mahavirj) [inline test_app/CMakeLists.txt:5]: This we can avoid for now to avoid CI failures with v4.3 release. We will have to see on how to make build configurable based on IDF version that component supports.

**Review by mahavirj — CHANGES_REQUESTED** [review]

File: test_app/CMakeLists.txt [hunk 11]
```diff
@@ -5,7 +5,7 @@ cmake_minimum_required(VERSION 3.5)
 set(EXTRA_COMPONENT_DIRS ../libsodium ../expat)
 
 # Set the components to include the tests for.
-set(TEST_COMPONENTS libsodium expat CACHE STRING "List of components to test")
+set(TEST_COMPONENTS libsodium expat esp_encrypted_img CACHE STRING "List of components to test")
```
Comment (mahavirj) [inline test_app/CMakeLists.txt:8]: Why this change is required? Can we remove it?

File: esp_encrypted_img/README.md [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+    } pre_enc_bin_header;
+
+The above struct represents encrypted image header.
+
+Note:
+* RSA-3072 key is provided to the tool externally. You can generate RSA key pair using following command:
+
+    `openssl genrsa -out rsa_key/private.pem 3072`
+
+* AES-GCM key is generated by the tool itself.
```
Comment (mahavirj) [inline esp_encrypted_img/README.md:28]:
```suggestion
* AES-GCM key and IV are generated by the tool itself.
```

**Review by mahavirj — CHANGES_REQUESTED** [review]

File: test_app/CMakeLists.txt (hunk 11 above)
Comment (tore-espressif) [reply test_app/CMakeLists.txt:8]: I'd vote for keeping it here. At least we are sure that the component compiles fine on all targets and supported IDF version.
Later we can run those tests too.

**Review by tore-espressif — APPROVED** [review]
LGTM!

**Review by mahavirj — APPROVED** [review]

**Comment by kumekay** [comment]
@tore-espressif  There was an issue with background processing, the component is published now https://components.espressif.com/component/espressif/esp_encrypted_img
